src/render-delaying-services.js

```diff
@@ -60,10 +89,25 @@ export function waitForServices(win) {
     return Services.timerFor(win).timeoutPromise(
         LOAD_TIMEOUT,
         getServicePromise(win, service),
-        `Render timeout waiting for service ${service} to be ready.`
+        `Render timeout waiting for service ${service} ` +
+      'to be ready.'
     );
   });
-  return Promise.all(promises);
+  return Promise.all(promises).then(services => {
+    const postInstallPromises = services.map(service => {
+      if (service && service.getPostRegisterRenderDelayPromise) {
+        return Services.timerFor(win).timeoutPromise(
+            POST_REGISTER_TIMEOUT,
+            service.getPostRegisterRenderDelayPromise(),
```

Comment:
How is service implementation getting canceled if it didn't get a chance to start?